pkg/module_manager/module.go

```diff
@@ -979,3 +1041,15 @@ func dumpData(filePath string, data []byte) error {
 	}
 	return nil
 }
+
+// ModuleProducer interface to get a factory which produces Modules CR object
+type ModuleProducer interface {
+	GetGVK() schema.GroupVersionKind
+	NewModule() ModuleObject
+}
+
+// ModuleObject Module CR
+type ModuleObject interface {
+	SetName(name string)
+	SetWeight(weight int)
+}
```

Comment:
Did I get it right that there is only an interface in the addon-operator, but the implementation of the interface is the duty of third-party users?

Reply:
Yes, you got it

Reply:
You can find an implementation [here](https://github.com/deckhouse/deckhouse/blob/154238fab9ad06321bd65b36f0abf04e67ae4c27/modules/002-deckhouse/hooks/pkg/apis/v1alpha1/module.go)